src/main/scala/util/RunUtils.scala

```diff
@@ -948,8 +967,51 @@ object RunUtils {
       writeToFile(newCFG.toDot(x => x.toString, Output.dotIder), s"${s}_resolvedCFG.dot")
     }
 
+    Logger.info("[!] Running Writes To")
+    val writesTo = WriteToAnalysis(ctx.program).analyze()
+    val reachingDefs = ReachingDefsAnalysis(ctx.program, writesTo).analyze()
+    config.analysisDotPath.foreach(
+      s =>
+        writeToFile(toDot(ctx.program), s"${s}_ct.dot")
+    )
+
+    Logger.info("[!] Running Symbolic Access Analysis")
+    val symResults: Map[CFGPosition, Map[SymbolicAccess, TwoElement]] =
+      SymbolicAccessAnalysis(ctx.program, analysisResult.last.IRconstPropResult).analyze()
+    config.analysisDotPath.foreach(s =>
+      writeToFile(toDot(ctx.program, symResults.foldLeft(Map(): Map[CFGPosition, String]) {
+        (m, t) =>
+          m + (t._1 -> t._2.toString)
+      }), s"${s}_saa.dot")
+    )
+
+
+    Logger.info("[!] Running Parameter Analysis")
+    val paramResults = ParamAnalysis(ctx.program).analyze()
+
+    Logger.info("[!] Running DSA Analysis")
+    val symbolTableEntries: Set[SymbolTableEntry] = ctx.globals ++ ctx.funcEntries
+    val dsa = DSA(ctx.program, symResults, analysisResult.last.IRconstPropResult, symbolTableEntries, ctx.globalOffsets, ctx.externalFunctions, reachingDefs, writesTo, paramResults)
+    dsa.analyze()
+
     Logger.info(s"[!] Finished indirect call resolution after $iteration iterations")
-    analysisResult.last
+    StaticAnalysisContext(
+      cfg = analysisResult.last.cfg,
+      constPropResult = analysisResult.last.constPropResult,
+      IRconstPropResult = analysisResult.last.IRconstPropResult,
+      memoryRegionResult = analysisResult.last.memoryRegionResult,
+      vsaResult = analysisResult.last.vsaResult,
+      interLiveVarsResults = analysisResult.last.interLiveVarsResults,
+      paramResults = paramResults, //analysisResult.last.paramResults,
+      steensgaardResults = analysisResult.last.steensgaardResults,
+      mmmResults = analysisResult.last.mmmResults,
+      memoryRegionContents = analysisResult.last.memoryRegionContents,
+      symbolicAccessess = symResults, // analysisResult.last.symbolicAccessess,
+      locals = Some(dsa.locals.toMap),
+      bus = Some(dsa.bu.toMap),
+      tds = Some(dsa.td.toMap),
+      reachingDefs = analysisResult.last.reachingDefs
+    )
```

Comment:
Why is this separate to the rest of the static analysis?

Reply:
it relies on resolved procedure calls

Reply:
Why does it rely on resolved indirect calls? This analysis is doing something similar enough to the Steensgaard analysis that is currently used to resolve indirect calls, so why does it need the indirect calls to be resolved first?

Reply:
DSA can resolve indirect calls, however, this functionality was not implemented. Due to the lack of type info from the lifter, SAA (symbolic access analysis) is used to find possible pointers. SAA is inter procedural so it requires call resolution first. SAA results are heavily used throughout DSA. Most of that info probably can be gathered through local reaching definitions (to find stack locations), and the symbol table (for globals) but removing SAA causes the analysis to lose some information about heap objects and unknowns (loaded from memory) . 

I think it's possible completely remove SAA and fully rely on DSA for call resolution but it is beyond this pull request

Reply:
Resolving indirect calls is necessarily going to be an iterative process, so I don't think SAA has to depend on indirect call resolution at all. DSA is ultimately a form of points-to analysis so it doesn't really make sense to make it depend on the Steensgaard points-to analysis. This doesn't matter for now but is worth further consideration when we try to actually do anything with the DSA results.